### PR TITLE
Build mappers from JSON and to JSON value

### PR DESCRIPTION
**Pull request by pakubieniec**
1. Create JSONMapper class (use Jackson JSON) which two methods:
 - mapJsonToJava ( )
 - mapObjectToJson ( )
2. TestMapper to check methods.
3. Update class FetchToTheForexApi
 - getRespondsWithHistoricalDateObj () = method return DataFromForexApi Object
4. Create class DataFromForexApi - includes all variables needs to fetching values ​​from API

### CODE REVIEW

File: src/main/java/DataModel/ExchangeRate.java [hunk 1]
```diff
@@ -4,18 +4,28 @@
 
 public class ExchangeRate {
     private String ticker;
+    private String date;
```
Comment (Kalololol) [inline src/main/java/DataModel/ExchangeRate.java:7]: Why String?

File: src/main/java/org/example/DataFromForexApi.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
+package org.example;
+
+import java.util.Objects;
+
+public class DataFromForexApi {
```
Comment (Kalololol) [inline src/main/java/org/example/DataFromForexApi.java]: The class has methods "set", why? maybe read-only?

File: src/main/java/services/FetchToTheForexAPI.java [hunk 3]
```diff
@@ -12,6 +12,7 @@
 public class FetchToTheForexAPI {
```
Comment (Kalololol) [inline src/main/java/services/FetchToTheForexAPI.java:11]: I think than API with nbp is easier :)
https://api.nbp.pl/

File: src/test/java/TestJsonMapper.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
+import Mappers.JSONMapper;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import services.FetchToTheForexAPI;
+
+import java.nio.channels.AsynchronousServerSocketChannel;
+import java.util.Objects;
+
+public class TestJsonMapper {
```
Comment (Kalololol) [inline src/test/java/TestJsonMapper.java:9]: If String Json = null or maybe "abc"?

File: src/main/java/services/FetchToTheForexAPI.java [hunk 5]
```diff
@@ -39,6 +49,17 @@ private String getResponds(HttpRequest request) {
...
 
+    private DataFromForexApi getRespondsObj(HttpRequest request) {
+        DataFromForexApi dataFromForexApi;
+        HttpResponse<String> response;
+        try {
+            response = CLIENT.send(request, HttpResponse.BodyHandlers.ofString());
+        } catch (IOException | InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+        dataFromForexApi = jsonMapper.mapJsonToJava(response.body());
```
Comment (Kalololol) [inline src/main/java/services/FetchToTheForexAPI.java]: return jsonMapper.mapJsonToJava(response.body());

**Review by Kalololol — COMMENTED** [review]
I think it's well written

File: src/main/java/DataModel/ExchangeRate.java (hunk 1 above)
Comment (pakubieniec) [reply src/main/java/DataModel/ExchangeRate.java:7]: That was my first thought.

File: src/main/java/org/example/DataFromForexApi.java (hunk 2 above)
Comment (pakubieniec) [reply src/main/java/org/example/DataFromForexApi.java]: You're right. Good idea.

File: src/main/java/services/FetchToTheForexAPI.java (hunk 3 above)
Comment (pakubieniec) [reply src/main/java/services/FetchToTheForexAPI.java:11]: Possible, but I had to use a different API

File: src/test/java/TestJsonMapper.java (hunk 4 above)
Comment (pakubieniec) [reply src/test/java/TestJsonMapper.java:9]: I see I need to do an additional test :)

File: src/main/java/services/FetchToTheForexAPI.java (hunk 5 above)
Comment (pakubieniec) [reply src/main/java/services/FetchToTheForexAPI.java]: Thanks!!!

**Pull request merged**